src/ui/src/lib/firmware.ts

```
@@ -111,16 +112,28 @@ export const releases = readable<Map<string, Release[]>>(new Map(), function sta
 	};
 });
 
-export function getFirmwareUrl(updateMethod: string, version: string, artifact: string, firmware: string): string {
+export function getFirmwareUrl(firmwareSource: string, version: string, artifact: string, firmware: string): string | null {
 	if (firmware)
-		switch (updateMethod) {
+		switch (firmwareSource) {
 			case 'artifact':
 				if (artifact) return `https://nightly.link/ESPresense/ESPresense/actions/runs/${artifact}/${firmware}.zip`;
```

Biome (1.9.4)

[error] This case is falling through to the next case.

Add a `break` or `return` statement to the end of this case to prevent fallthrough.

(lint/suspicious/noFallthroughSwitchClause)